### ISSUE TITLE
Update GRUB to address recent CVE-2020-10713 BootHole security vulnerability (and more) - 29 Jul 2020

### ISSUE THREAD

**Issue by stefano-m** [NixOS/nixpkgs#94224]
**Describe the bug**

https://www.debian.org/security/2020/dsa-4735

Several vulnerabilities have been discovered in the GRUB2 bootloader.

-    CVE-2020-10713 A flaw in the grub.cfg parsing code was found allowing to break UEFI Secure Boot and load arbitrary code. Details can be found at https://www.eclypsium.com/2020/07/29/theres-a-hole-in-the-boot/
-    CVE-2020-14308 It was discovered that grub_malloc does not validate the allocation size allowing for arithmetic overflow and subsequently a heap-based buffer overflow.
-    CVE-2020-14309 An integer overflow in grub_squash_read_symlink may lead to a heap based buffer overflow.
-    CVE-2020-14310 An integer overflow in read_section_from_string may lead to a heap based buffer overflow.
-    CVE-2020-14311 An integer overflow in grub_ext2_read_link may lead to a heap-based buffer overflow.
-    CVE-2020-15706  script: Avoid a use-after-free when redefining a function during execution.
-    CVE-2020-15707 An integer overflow flaw was found in the initrd size handling.

**To Reproduce**

N/A

**Screenshots**
N/A

**Additional context**
N/A

**Notify maintainers**

No maintainers listed in `grub2.meta`

Cc @grahamc of NixOS security team

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.50-hardened, NixOS, 20.03.2425.9a3e34ae601 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"nixos-20.03.2611.bb8f0cc2279"`
 - channels(user): `"unstable-20.09pre235212.a5cc7d31977"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: [ grub2 ]
```

**Comment by zzywysm** [comment]
I'm not sure how much of a priority this is, given that NixOS doesn't support Secure Boot in the first place:

https://github.com/NixOS/nixpkgs/issues/42127

**Comment by xfix** [comment]
Yeah, uh, https://bugzilla.redhat.com/show_bug.cgi?id=1861977. Considering NixOS doesn't support Secure Boot, we are better of not applying the patch at least in the current state.

**Comment by stefano-m** [comment]
Although maybe not as important as for systems where SecureBoot works, please be aware that CVE-2020-10713 isn't the only vulnerability that has been fixed.

Also, I noticed that [debian fixed a regression in grub](https://security-tracker.debian.org/tracker/DSA-4735-2) because

> The update for grub2 released as DSA 4735-1 caused a boot-regression
when chainloading another bootlaoder and breaking notably dual-boot with
Windows. Updated grub2 packages are now available to correct this issue.

although it might not be the same issue as Red Hat's (which [seems to have been now fixed](https://bugzilla.redhat.com/show_bug.cgi?id=1861977#c57)).

**Comment by xfix** [comment]
> Although maybe not as important as for systems where SecureBoot works, please be aware that CVE-2020-10713 isn't the only vulnerability that has been fixed.

The rest of the issues are also only meaningful to exploit with Secure Boot. Without Secure Boot an attacker could easily substitute the boot loader with their code instead of bothering with exploiting vulnerabilities within GRUB.

**Comment by worldofpeace** [comment]
Has this been fixed?
cc @NixOS/nixos-release-managers

**Comment by xfix** [comment]
No, and I don't think we should fix those issues. The patches are still buggy (see https://bugzilla.redhat.com/show_bug.cgi?id=1861977), this is bootloader (which is non-trivial to rollback), and the vulnerabilities have 0 impact without Secure Boot.

**Comment by stefano-m** [comment]
Should this be a "wontfix" due to the above reasons then?

**Comment by 06kellyjac** [comment]
Maybe but there is a WIP PR working on Secure Boot #53901

**Comment by xfix** [comment]
This particular PR is for systemd-boot.

**Comment by 06kellyjac** [comment]
ah, whoops

**Comment by xfix** [comment]
I think that may have been fixed by #115310.
